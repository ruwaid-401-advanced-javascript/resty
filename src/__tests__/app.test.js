import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Form from '../components/form';
import Header from '../components/header';
import Footer from '../components/footer';

describe('<routes/>', () => {
  it('is alive at application start', () => {
    let app = shallow(<Header />);
    expect(app.find('h1').exists()).toBeTruthy();
  });

  it('is alive at application start', () => {
    let app = shallow(<Footer />);
    expect(app.find('footer').exists()).toBeTruthy();
  });

  it('is alive at application start', () => {
    let app = shallow(<Form />);
    expect(app.find('button').exists()).toBeTruthy();
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Form />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});