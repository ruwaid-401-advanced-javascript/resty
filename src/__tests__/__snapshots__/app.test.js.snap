// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> renders correctly 1`] = `
<div
  id="form"
>
  <div
    id="main-header"
  >
    <div
      id="url-div"
    >
      <label
        for="url"
      >
        URL:
      </label>
      <input
        id="url"
      />
      <button
        onClick={[Function]}
      >
        GO!
      </button>
      <div
        id="header"
      >
        <label>
           Header
        </label>
        <br />
        <label
          for="Basic"
        >
          Basic
        </label>
        <input
          checked={true}
          id="Basic"
          name="auth"
          onChange={[Function]}
          type="radio"
          value="Basic"
        />
        <label
          for="Bearer"
        >
          Bearer
        </label>
        <input
          id="Bearer"
          name="auth"
          onChange={[Function]}
          type="radio"
          value="Bearer"
        />
        <br />
        <input
          id="headerValue"
          onChange={[Function]}
        />
      </div>
      <label
        id="textarea"
      >
        Body
        <textarea
          cols="50"
          onChange={[Function]}
          rows="4"
        />
      </label>
    </div>
    <div
      id="buttons"
    >
      <button
        class="aqua"
        id="GET"
        onClick={[Function]}
      >
        GET
      </button>
      <button
        id="POST"
        onClick={[Function]}
      >
        POST
      </button>
      <button
        id="PUT"
        onClick={[Function]}
      >
        PUT
      </button>
      <button
        id="DELETE"
        onClick={[Function]}
      >
        DELETE
      </button>
    </div>
  </div>
</div>
`;

exports[`<routes/> renders correctly 1`] = `
<div
  id="form"
>
  <div
    id="main-header"
  >
    <div
      id="url-div"
    >
      <label
        for="url"
      >
        URL:
      </label>
      <input
        id="url"
      />
      <button
        onClick={[Function]}
      >
        GO!
      </button>
      <div
        id="header"
      >
        <label>
           Header
        </label>
        <br />
        <label
          for="Basic"
        >
          Basic
        </label>
        <input
          checked={true}
          id="Basic"
          name="auth"
          onChange={[Function]}
          type="radio"
          value="Basic"
        />
        <label
          for="Bearer"
        >
          Bearer
        </label>
        <input
          id="Bearer"
          name="auth"
          onChange={[Function]}
          type="radio"
          value="Bearer"
        />
        <br />
        <input
          id="headerValue"
          onChange={[Function]}
        />
      </div>
      <label
        id="textarea"
      >
        Body
        <textarea
          cols="50"
          onChange={[Function]}
          rows="4"
        />
      </label>
    </div>
    <div
      id="buttons"
    >
      <button
        class="aqua"
        id="GET"
        onClick={[Function]}
      >
        GET
      </button>
      <button
        id="POST"
        onClick={[Function]}
      >
        POST
      </button>
      <button
        id="PUT"
        onClick={[Function]}
      >
        PUT
      </button>
      <button
        id="DELETE"
        onClick={[Function]}
      >
        DELETE
      </button>
    </div>
  </div>
</div>
`;
